src/app/ReviewService.Abstractions/IReviewService.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace ReviewService.Abstractions;
+
+/// <summary>
+/// Provides ways to validate review prompt conditions using <see cref="IReviewSettingsSource{TReviewPromptSettings}"/> and
+/// to prompt user to review the current application using <see cref="IReviewPrompter"/>.
+/// </summary>
+/// <typeparam name="TReviewSettings">The type of the object that we use for tracking.</typeparam>
+public interface IReviewService<TReviewSettings>
+	where TReviewSettings : ReviewSettings
+{
+	/// <summary>
+	/// Checks if all review prompt conditions are satisfied and then prompt user to review the current application.
+	/// </summary>
+	/// <param name="ct"><see cref="CancellationToken"/>.</param>
+	/// <returns><see cref="Task"/>.</returns>
+	Task TryRequestReview(CancellationToken ct);
+
+	/// <summary>
+	/// Tracks that the application was launched.
+	/// </summary>
+	/// <param name="ct">The cancellation token.</param>
+	/// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+	Task TrackApplicationLaunched(CancellationToken ct);
+
+	/// <summary>
+	/// Tracks that a primary action was completed.
+	/// </summary>
+	/// <param name="ct"><see cref="CancellationToken"/>.</param>
+	/// <returns><see cref="Task"/>.</returns>
+	Task TrackPrimaryActionCompleted(CancellationToken ct);
+
+	/// <summary>
+	/// Tracks that a secondary action was completed.
+	/// </summary>
+	/// <param name="ct"><see cref="CancellationToken"/>.</param>
+	/// <returns><see cref="Task"/>.</returns>
+	Task TrackSecondaryActionCompleted(CancellationToken ct);
+
+	/// <summary>
+	/// Updates the persisted <see cref="TReviewSettings"/>.
+	/// </summary>
+	/// <param name="ct"><see cref="CancellationToken"/>.</param>
+	/// <param name="updateFunction">Function that returns updated <see cref="TReviewSettings"/>.</param>
+	/// <returns><see cref="Task"/>.</returns>
+	Task UpdateReviewSettings(CancellationToken ct, Func<TReviewSettings, TReviewSettings> updateFunction);
+
+	/// <summary>
+	/// Resets the persisted <see cref="TReviewSettings"/>.
+	/// </summary>
+	/// <param name="ct"><see cref="CancellationToken"/>.</param>
+	/// <returns><see cref="Task"/>.</returns>
+	Task ResetReviewSettings(CancellationToken ct);
```

Comment:
I'm not sure to see the use-case for this method.

Reply:
Thought we could reset review for an application after an update could be useful but I wasn't even sure so let's remove it?